core-bundle/src/HttpKernel/ModelArgumentResolver.php

```diff
@@ -79,6 +80,15 @@ private function fetchModel(Request $request, ArgumentMetadata $argument): ?Mode
             return $value;
         }
 
+        // Special handling for pageModel that could be globally registered
+        if (
+            isset($GLOBALS['objPage'])
+            && $GLOBALS['objPage'] instanceof PageModel
+            && $GLOBALS['objPage']->id === (int) $value
```

Comment:
Afaik the id value is a string, so this comparision is very likely never true. I suggest to use `(int) $GLOBALS['objPage']->id === (int) $value`

Reply:
Fixed in a5c2326300d31a03b92cc81f12cab5b9fceaa9bc